google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -1572,6 +1597,15 @@ void markUsed() {
       lastUseTime = clock.instant();
     }
 
+    /**
+     * Method to mark a session that is to be occupied by a possibly long-running transaction. Any
+     * transaction that is expected to be long-running (for ex - Partitioned DML, Batch Read) must
+     * use this method.
+     */
+    void markEligibleForLongRunning() {
+      eligibleForLongRunning = true;
+    }
```

Comment:
Thanks for pointing out. I'll try to explain the small differences here.

* Yeah, I have added a method since there can be more than one transaction types (PDML, Batch Reads) that can be eligible for long running. And hence, it will be cleaner for all such transaction types to call a single method. Today, BatchReads don't use the session pool hence it looks as if PDML is the only type that calls `markEligibleForLongRunning` . But in general the method can be used by other transaction types.
* We don't need a method `eligibleForLongRunning = false` since its a common logic that's present in two places for all transactions a) during initialisation and b) during Session#close() method. Transactions don't need to set this flat to false.